internal/resolvers/types.go

```diff
@@ -0,0 +1,11 @@
+package resolvers
+
+import "github.com/bluesky-social/indigo/atproto/crypto"
+
+type HabitatHostResolver func(did string) (habitatHost string, err error)
+
+type PublicKeyResolver func(did string) (publicKey crypto.PublicKey, err error)
+
+type PrivateKeyResolver func(did string) (privateKey crypto.PrivateKey, err error)
+
+// TODO resolve the DID from a host
```

Comment:
doesn't atproto PDS do this already?

Reply:
Not really related to this PR, but it's something you need when doing OAuth, and I had a helper for it there. It would make sense to factor them out here (assuming OAuth was not an external package)